merge.go

```diff
@@ -163,6 +165,50 @@ func (m *Merger) mergeArray(orig []interface{}, n []interface{}, node string) []
 		}
 
 		return m.mergeArrayByKey(orig, n[1:], node, key)
+
+	} else if should, idx := shouldInsertIntoArrayBasedOnIndex(n); should {
+		if idx < 0 || idx > len(orig) {
+			m.Errors.Append(ansi.Errorf("@m{%s}: @R{specified insertion index} @c{%d} @R{is out of bounds}", node, idx))
+			return nil
+		}
+
+		DEBUG("%s: inserting %d new elements to existing array, at index %d", node, len(n)-1, idx)
+		return append(orig[0:idx], append(n[1:], orig[idx:]...)...)
+
+	} else if should, relative, key, name := shouldInsertIntoArrayBasedOnName(n); should {
```

Comment:
This might need to be swapped, to insert based on name before insert based on index, to handle cases where the element has a name/key/id value that is numeric. Can you verify that with tests?